platform/android/src/file_source.cpp

```diff
@@ -63,11 +63,18 @@ void FileSource::setResourceTransform(jni::JNIEnv& env, jni::Object<FileSource::
 }
 
 void FileSource::resume(jni::JNIEnv&) {
-    fileSource->resume();
+    counter++;
+    if (counter==1 && wasPaused) {
```

Comment:
Do we need to check both `counter` and `wasPaused`? The latter seems redundant.

Reply:
Also, spaces (`==`)

Reply:
it seems that it's only valid to call resume if the filesource was paused first (else the assertion will fail in core). I reworked the code to use remove wasPaused and wrapped the counter in an optional instead.